specs/tree.spec.js

```diff
@@ -0,0 +1,145 @@
+const Tree = require('../src/Tree');
+
+describe('Tree', function() {
+    describe('Constructor', function() {
+        it('New constructor should create a root equal null', function() {
+            const tree = new Tree();
+            expect(tree.root).to.be.equal(null);
+        });
+    });
+    describe('insert', function() {
+        it('Should set key, value to root if root is null', function() {
+            const tree = new Tree();
+            tree.insert('elem1', 1);
+            expect(tree.root.key).to.be.equal('elem1');
+        });
+        it('Should set key, value to root if root is null', function() {
+            const tree = new Tree();
+            tree.insert('elem1', 1);
+            expect(tree.root.value).to.be.equal(1);
+        });
+        it('Should set key to left child node if node.key less that parent.key and parent.left child is empty', function() {
+            const tree = new Tree();
+            tree.insert(7, 'elem1');
+            tree.insert(1, 'elem2')
+            expect(tree.root.left.key).to.be.equal(1);
+            expect(tree.root.left.value).to.be.equal('elem2');
+        });
+        it('Should set key to correct place in tree structure', function() {
+            const tree = new Tree();
+            tree.insert(7, 'elem1');
+            tree.insert(4, 'elem2')
+            tree.insert(2, 'elem3');
+            tree.insert(5, 'elem4')
+            tree.insert(9, 'elem5');
+            tree.insert(1, 'elem6')
+            expect(tree.root.left.left.key).to.be.equal(2);
+            expect(tree.root.left.value).to.be.equal('elem3');
+        });
+        it('Should set key to correct place in tree structure', function() {
+            const tree = new Tree();
+            tree.insert(7, 'elem1');
+            tree.insert(4, 'elem2')
+            tree.insert(2, 'elem3');
+            tree.insert(5, 'elem4')
+            tree.insert(9, 'elem5');
+            tree.insert(1, 'elem6')
+            expect(tree.root.left.right.key).to.be.equal(5);
+            expect(tree.root.left.right.value).to.be.equal('elem4');
+        });
+        it('Should set key to correct place in tree structure', function() {
+            const tree = new Tree();
+            tree.insert(7, 'elem1');
+            tree.insert(4, 'elem2')
+            tree.insert(2, 'elem3');
+            tree.insert(5, 'elem4')
+            tree.insert(9, 'elem5');
+            tree.insert(1, 'elem6')
+            expect(tree.root.right.left.key).to.be.equal(1);
+            expect(tree.root.right.left.key).to.be.equal('elem6');
+        });
```

Comment:
3 redundant tests. better to separate them onto 3 with more descriptive names and scenarios.